### PR TITLE
Remove CRDs during uninstallation of owner component

### PR DESCRIPTION
**Pull request by nikhil-thomas**
Remove CRD fliter from uninstall mechanism

Signed-off-by: Nikhil Thomas <nikthoma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by pradeepitm12 — COMMENTED** [review]
/lgtm

**Review by savitaashture — APPROVED** [review]

File: pkg/reconciler/common/install.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ func Install(ctx context.Context, manifest *mf.Manifest, instance v1alpha1.Tekto
 
 // Uninstall removes all resources except CRDs, which are never deleted automatically.
```
Comment (sthaha) [inline pkg/reconciler/common/install.go:74]: Is this correct anymore?

File: pkg/reconciler/common/install_test.go [hunk 2]
```diff
@@ -140,10 +140,9 @@ func TestUninstall(t *testing.T) {
 	crd := clusterScopedResource("apiextensions.k8s.io/v1beta1", "CustomResourceDefinition", "test-crd")
 
 	// Deliberately mixing the order in the manifest.
-	in := []unstructured.Unstructured{deployment, role, roleBinding, clusterRole, clusterRoleBinding, crd}
+	in := []unstructured.Unstructured{crd, deployment, role, roleBinding, clusterRole, clusterRoleBinding}
```
Comment (sthaha) [inline pkg/reconciler/common/install_test.go:143]: why is this change required?

**Review by sthaha — COMMENTED** [review]
implementation looks good to me but I think the comments should be corrected as well to align with this change.

File: pkg/reconciler/common/install_test.go (hunk 2 above)
Comment (nikhil-thomas) [reply pkg/reconciler/common/install_test.go:143]: he deletion happens back to front, so the ordered will be reveresed.
earlier the pass condition is that crds will not be deleted. then the order where we put the crd in the first list did not matter, as crd will not be in the deletes list

**Review by sthaha — COMMENTED** [review]
/lgtm
/approve
